pkg/controller/servicebinding/secret_test.go

```diff
@@ -38,17 +36,98 @@ func TestSecretNew(t *testing.T) {
 		f.FakeDynClient(),
 		ns,
 		name,
+		data,
+		secretOwnerReference,
 	)
 
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName, err := HashedResourceNameForData(name, data)
+		assert.NoError(t, err)
+		assert.Contains(t, hashedName, name)
+	})
+
 	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+		u, err := s.createOrUpdate()
 		assert.NoError(t, err)
+		hashedName, err := HashedResourceNameForData(name, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, u.GetName())
 		assertSecretNamespacedName(t, u, ns, name)
 	})
 
 	t.Run("Get", func(t *testing.T) {
 		u, err := s.get()
 		assert.NoError(t, err)
+		hashedName, err := HashedResourceNameForData(name, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName, u.GetName())
 		assertSecretNamespacedName(t, u, ns, name)
 	})
+
+	t.Run("different_secret_payload_have_different_name", func(t *testing.T) {
+		payload1 := map[string][]byte{"dbName": []byte("postgres")}
+		value1 := newSecret(
+			f.FakeDynClient(),
+			ns,
+			name,
+			payload1,
+			secretOwnerReference,
+		)
+		secret1, err := value1.createOrUpdate()
+		assert.NoError(t, err)
+
+		payload2 := map[string][]byte{"user": []byte("admin")}
+		value2 := newSecret(
+			f.FakeDynClient(),
+			ns,
+			name,
+			payload2,
+			secretOwnerReference,
+		)
+		secret2, err := value2.createOrUpdate()
+		assert.NoError(t, err)
+
+		assert.NotEqual(t, secret1.GetName(), secret2.GetName())
+	})
+
+	t.Run("secret_with_empty_payload", func(t *testing.T) {
```

Comment:
What is the expectation? What should the `GetName()` return when the payload is empty?

Reply:
If the payload is empty then the secret name is same as `ServiceBinding` name.

Reply:
but the assertions in the tests later on say the contrary.